core/templates/dev/head/pages/exploration-player-page/layout-directives/learner-view-info.directive.ts

```diff
@@ -35,24 +37,30 @@ angular.module('oppia').directive('learnerViewInfo', [
     return {
       restrict: 'E',
       scope: {},
-      bindToController: {
-        explorationTitle: '@'
-      },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration-player-page/layout-directives/' +
         'learner-view-info.directive.html'),
       controllerAs: '$ctrl',
       controller: [
         '$http', '$log', '$uibModal', 'ContextService',
-        'UrlInterpolationService', 'DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR',
+        'ReadOnlyExplorationBackendApiService', 'UrlInterpolationService',
+        'UrlService', 'DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR',
         'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
         function($http, $log, $uibModal, ContextService,
-            UrlInterpolationService, DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR,
+            ReadOnlyExplorationBackendApiService, UrlInterpolationService,
+            UrlService, DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR,
             EXPLORATION_SUMMARY_DATA_URL_TEMPLATE) {
           var ctrl = this;
           var explorationId = ContextService.getExplorationId();
           var expInfo = null;
 
+          ctrl.explorationTitle = 'Loading...';
+          ReadOnlyExplorationBackendApiService.fetchExploration(
+            explorationId, UrlService.getExplorationVersionFromUrl())
```

Comment:
This will make a call irrespective of whether we already have fetched the entire exploration, right?

Don't we have any other services which hold this data?

Reply:
The ReadOnlyBackendApiService also caches the exploration, but it caches only the latest exploration.

Reply:
Can you please check how many calls we are making while loading exploration player? (Maybe through dev tools.)

Reply:
@DubeySandeep 7 ajax requests are made to `/exploration_handler/init`

Reply:
I think the reason for this is that the exploration data is what is cached instead of the response promise.

Reply:
So if 2 requests are made to the ReadOnlyBackendApiService, the result will only get cached after the request is completed. But if the promise was cached, then only one request will be made and subsequent calls will be served with the promise, and will all get resolved at the same time

Reply:
I think it would make sense to cache promise instead of data everywhere we can.

Reply:
Yes, but I don't think that should be done in this PR since this PR is needed for this release.

Reply:
Yes, create an issue for this please, once the issue is created we can merge this.

Reply:
#7504